apis/infrastructure/v1beta1/byocluster_webhook_test.go

```diff
@@ -0,0 +1,104 @@
+
+var _ = Describe("ByoclusterWebhook", func() {
+	Context("When ByoCluster gets a create request", func() {
+		var (
+			byoCluster        *ByoCluster
+			ctx               context.Context
+			k8sClientUncached client.Client
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+			var clientErr error
+			k8sClientUncached, clientErr = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+			Expect(clientErr).NotTo(HaveOccurred())
+
+			byoCluster = &ByoCluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ByoCluster",
+					APIVersion: clusterv1.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "byocluster-create",
+					Namespace: "default",
+				},
+				Spec: ByoClusterSpec{},
+			}
+		})
+
+		It("should reject the request when BundleLookupTag is empty", func() {
+			err := k8sClientUncached.Create(ctx, byoCluster)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError("admission webhook \"vbyocluster.kb.io\" denied the request: ByoCluster.infrastructure.cluster.x-k8s.io \"" + byoCluster.Name + "\" is invalid: <nil>: Internal error: cannot create ByoCluster without Spec.BundleLookupTag"))
+		})
+
+		It("should success when BundleLookupTag is not empty", func() {
+			byoCluster.Spec.BundleLookupTag = "v0.1.0_alpha.2"
+			err := k8sClientUncached.Create(ctx, byoCluster)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+	})
+
+	Context("When ByoCluster gets an update request", func() {
+		var (
+			byoCluster        *ByoCluster
+			ctx               context.Context
+			k8sClientUncached client.Client
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+			var clientErr error
+			k8sClientUncached, clientErr = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+			Expect(clientErr).NotTo(HaveOccurred())
+
+			byoCluster = &ByoCluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ByoCluster",
+					APIVersion: clusterv1.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "byocluster-update",
+					Namespace: "default",
+				},
+				Spec: ByoClusterSpec{
+					BundleLookupTag: "v0.1.0_alpha.2",
+				},
+			}
+
+			Expect(k8sClientUncached.Create(ctx, byoCluster)).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			Expect(k8sClientUncached.Delete(ctx, byoCluster)).Should(Succeed())
+		})
+
+		It("should reject the request when BundleLookupTag is empty", func() {
+			byoCluster.Spec.BundleLookupTag = ""
+			err := k8sClientUncached.Update(ctx, byoCluster)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError("admission webhook \"vbyocluster.kb.io\" denied the request: ByoCluster.infrastructure.cluster.x-k8s.io \"" + byoCluster.Name + "\" is invalid: <nil>: Internal error: cannot update ByoCluster with empty Spec.BundleLookupTag"))
+		})
+
+		It("should success when BundleLookupTag is not empty", func() {
+			byoCluster.Spec.BundleLookupTag = "fake"
+			err := k8sClientUncached.Update(ctx, byoCluster)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
How about getting the cluster object and then checking if value got updated?

Reply:
ok